### PR TITLE
Fix bug 1148820: Properly get the summary for translations in jsons

### PR DESCRIPTION
**Pull request by Elchi3**
https://bugzilla.mozilla.org/show_bug.cgi?id=1148820

### CODE REVIEW

**Comment by groovecoder** [comment]
I'll take a look at this but I'm going to take some time to be extra careful. Calling methods like `get_summary()` from within other methods like `build_json_task` have introduced infinitely recursive loops in the past. Unless this simply reverts a previous change?

**Comment by Elchi3** [comment]
Nope this is not reverting. https://github.com/mozilla/kuma/pull/2524 introduced this wrongly with the translation summary not working. The code is now doing the same for the translation summary as it is for the en-US summary in the json. Whether or not that is performant, I don't know.

File: kuma/wiki/models.py [hunk 1]
```diff
@@ -568,6 +568,10 @@ def build_json_data(self):
         if self.pk:
             for translation in self.other_translations:
                 revision = translation.current_revision
+                if revision.summary:
+                   summary = revision.summary
+                else:
+                   summary = translation.get_summary(strip_markup=False)
```
Comment (groovecoder) [inline kuma/wiki/models.py]: Below, we do a `if self.current_revision.summary` check before we call `self.get_summary()`. I notice the main change here is to remove it?

Could we uplift the same check here? I.e., `if translation.current_revision: summary = translation.current_revision.summary; else summary = translation.get_summary()`? I'd just like to call `get_summary()` as little as possible.

Comment (Elchi3) [inline kuma/wiki/models.py]: line 571 is exactly that check, no?

Comment (groovecoder) [inline kuma/wiki/models.py]: oh, duh. I was just looking at the green lines so I ignored that 570 sets `revision` to `translation.current_revision`

**Pull request merged**